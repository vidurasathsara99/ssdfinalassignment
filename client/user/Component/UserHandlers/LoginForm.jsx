import React from "react";
import {UserRoles} from "../../api/roles";
import resources from "../../resource.config";
import Cookies from 'js-cookie';
import Footer from "../Footer";

export default class LoginForm extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            server_msg:null
        }
        this.redirectToSignUp=this.redirectToSignUp.bind(this);
        this.handleLogin=this.handleLogin.bind(this);
    }
    redirectToSignUp(){
        window.location.href = "/sign-up";
    }

    async handleLogin(){
        const username = this.username.value;
        let loginData = {"username":username,"password":this.password.value};
        await fetch(resources.proxy("/login"),{
            method:'post',
            headers: {'Content-Type':'application/json'},
            body: JSON.stringify(loginData)
        }).then(r=>r.text()).then(d=>this.setState({server_msg:d})).catch(e=>console.log(e));
        const server_msg = this.state.server_msg;
        if(server_msg.substring(0,7)==="success"){
            const id = server_msg.substring(8,server_msg.length);
            let prefix = username.charAt(0);
            Cookies.set('username',username);
            Cookies.set('userid',id);
            Cookies.set('jwt_token', 'HeuteIstDerErsteTagVomRestDeinesLebens');
            switch (prefix){
                case 'S': window.location.href="/admin"; break;
                case 'E': window.location.href="/editor"; break;
                case 'C': window.location.href="/reviewer"; break;
                case 'R': window.location.href="/user"; break;
                case 'W': window.location.href="/worker"; break;
                case 'A': window.location.href="/user"; break;
                default: this.setState({server_msg:"Unknown user!"});
            }
        }
    }

    render() {


        const server_msg = this.state.server_msg;
        let error_msg;
        if(server_msg!==null)
            if(server_msg.length>8)
                if(server_msg.substring(0,7)!=="success")
                    error_msg = server_msg;
        return <React.Fragment>

            <center>
            <div className={"card w-25 d-flex justify-content-center"} style={{backgroundColor: "#F8EEFF"}}>
                <div className={"card-body"}>


                    <h3 style={{color:"purple"}}>Login</h3>
            <p>{error_msg}</p>

                <div className="form-group">
            <input type={"text"} class="form-control" placeholder={"Enter username"} ref={(ref) => {this.username = ref}}/>
            <p/>
            <input type={"password"}  class="form-control"  placeholder={"Enter password"} ref={(ref) => {this.password = ref}}/>
            <p/>
</div>

            <button class="btn btn-success" onClick={this.handleLogin}>&nbsp;&nbsp;Login &nbsp;&nbsp;</button> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
{/*             <button class="btn btn-primary" onClick={this.redirectToSignUp}>&nbsp;&nbsp;Register &nbsp;&nbsp;</button> <br></br> */}
            <br/>
            <a href={"/forgot-password"}>Forgot Password</a>

                </div>
            </div>
            <div style={{position:"absolute",bottom:0, width:"100%", left:0}}>
                <Footer/>
            </div>
            </center>
        </React.Fragment>
    }
}
