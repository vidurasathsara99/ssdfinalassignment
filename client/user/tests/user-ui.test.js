import jest from 'jest';
import React from 'react';
import renderer from 'react-test-renderer';

import LoginForm from "../Component/UserHandlers/LoginForm";

test('Login Form rendered properly',()=>{
    const loginform = renderer.create(<LoginForm/>);
    let tree = loginform.toJSON();
    expect(tree).toMatchSnapshot();
});