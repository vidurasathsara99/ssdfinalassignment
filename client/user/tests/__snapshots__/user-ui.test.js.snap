// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form rendered properly 1`] = `
<center>
  <div
    className="card w-25 d-flex justify-content-center"
    style={
      Object {
        "backgroundColor": "#F8EEFF",
      }
    }
  >
    <div
      className="card-body"
    >
      <h3
        style={
          Object {
            "color": "purple",
          }
        }
      >
        Login
      </h3>
      <p />
      <div
        className="form-group"
      >
        <input
          class="form-control"
          placeholder="Enter username"
          type="text"
        />
        <p />
        <input
          class="form-control"
          placeholder="Enter password"
          type="password"
        />
        <p />
      </div>
      <button
        class="btn btn-success"
        onClick={[Function]}
      >
          Login   
      </button>
               
      <button
        class="btn btn-primary"
        onClick={[Function]}
      >
          Register   
      </button>
       
      <br />
      <br />
      <a
        href="/forgot-password"
      >
        Forgot Password
      </a>
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.05)",
          "borderRadius": "10px",
          "bottom": "0px",
          "left": "-3px",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        className="text-center p-3 mb-1"
      >
        <button
          className="btn btn-outline-success"
          onClick={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          UP
        </button>
         © 2021 Sri Lanka Institute of Information Technology, developed by 
        <b>
          EagleEyes
        </b>
         on behalf of AF project, All Rights Reserved
      </div>
    </div>
  </div>
</center>
`;
